tests/ext-monitor.test.js

```diff
@@ -243,10 +243,13 @@ test('listeners are registered at initialization', () => {
   };
 
   const extMonitor = new ExtensionMonitor();
+  const messageListenerFn = jest.spyOn(extMonitor, 'messageListener');
+  const onRemovedListenerFn = jest.spyOn(extMonitor, 'onRemovedListener');
+
   extMonitor.init();
 
-  // runtime.onMessage.addListener and tabs.onRemoved.addListener are called.
-  expect(addListener).toHaveBeenCalledTimes(2);
+  expect(addListener).toHaveBeenNthCalledWith(1, messageListenerFn);
+  expect(addListener).toHaveBeenNthCalledWith(2, onRemovedListenerFn);
```

Comment:
@rpl how is this approach to you ?

Reply:
> @rpl how is this approach to you ?
> **update**:
> I have checked like the following:
> 
> ```
> expect(addListener).toHaveBeenCalledTimes(2);
> expect(addListener).toHaveBeenNthCalledWith(1, messageListenerFn);
> expect(addListener).toHaveBeenNthCalledWith(2, onRemovedListenerFn);
> ```

Even with that change, the test does still pass if ext-monitor's `init` method does call (by mistake) twice `browser.tabs.onRemoved.addListener` on both the functions instead of calling `browser.runtime.onMessage.addListener` with `messageListenerFn` and `browser.tabs.onRemoved.addListener` with `onRemovedListenerFn`.

In practice, if ext-monitor's `init` does look like the following then the test case is supposed to fail:
```
  init() {
    browser.tabs.onRemoved.addListener(this.messageListener);
    browser.tabs.onRemoved.addListener(this.onRemovedListener);
  }
```

The current version doesn't.

If you just use two separate mocks for the two `addListener` methods, the test would fail as expected:
```
  const runtimeMessageAddListener = jest.fn();
  const tabsRemovedAddListener = jest.fn();

  window.browser = {
    runtime: { onMessage: { addListener: runtimeMessageAddListener } },
    tabs: { onRemoved: { addListener: tabsRemovedAddListener } },
  };

....
```